core/src/main/java/com/sequenceiq/cloudbreak/service/image/ImageProvider.java

```diff
@@ -1,19 +1,73 @@
 package com.sequenceiq.cloudbreak.service.image;
 
+import java.io.IOException;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Component;
 
 import com.sequenceiq.cloudbreak.cloud.model.catalog.CloudbreakImageCatalogV3;
 import com.sequenceiq.cloudbreak.cloud.model.catalog.Image;
 import com.sequenceiq.cloudbreak.core.CloudbreakImageNotFoundException;
+import com.sequenceiq.cloudbreak.service.stack.InstanceMetaDataService;
 
 @Component
 public class ImageProvider {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(ImageProvider.class);
+
+    @Inject
+    private ImageService imageService;
+
+    @Inject
+    private InstanceMetaDataService instanceMetaDataService;
+
     public Image getCurrentImageFromCatalog(String currentImageId, CloudbreakImageCatalogV3 imageCatalog)
             throws CloudbreakImageNotFoundException {
         return imageCatalog.getImages().getCdhImages().stream()
                 .filter(img -> currentImageId.equals(img.getUuid()))
                 .findFirst()
                 .orElseThrow(() -> new CloudbreakImageNotFoundException(String.format("Image not found with id: %s", currentImageId)));
     }
+
+    public com.sequenceiq.cloudbreak.cloud.model.Image getCurrentImageForUpgrade(Long stackId) throws CloudbreakImageNotFoundException {
+        com.sequenceiq.cloudbreak.cloud.model.Image currentImageFromStack = imageService.getImage(stackId);
+        Set<com.sequenceiq.cloudbreak.cloud.model.Image> imagesFromInstances = getImagesFromInstanceMetadata(stackId);
+        com.sequenceiq.cloudbreak.cloud.model.Image currentImage = getCurrentImage(currentImageFromStack, imagesFromInstances);
```

Comment:
I would call it something different to avoid confusion with the current images. Something that says different than the current image used for the stack.